sycl/source/detail/pi.cpp

```diff
@@ -37,22 +38,55 @@ std::string platformInfoToString(pi_platform_info info) {
 // Check for manually selected BE at run-time.
 bool useBackend(Backend TheBackend) {
   static const char *GetEnv = std::getenv("SYCL_BE");
-  static const Backend Use =
-    std::map<std::string, Backend>{
-      { "PI_OPENCL", SYCL_BE_PI_OPENCL },
-      { "PI_OTHER",  SYCL_BE_PI_OTHER }
+  static const Backend Use = std::map<std::string, Backend>{
+      {"PI_OPENCL", SYCL_BE_PI_OPENCL}, {"PI_OTHER", SYCL_BE_PI_OTHER}
       // Any other value would yield PI_OPENCL (current default)
-    }[ GetEnv ? GetEnv : "PI_OPENCL"];
+  }[GetEnv ? GetEnv : "PI_OPENCL"];
   return TheBackend == Use;
 }
 
 // Definitions of the PI dispatch entries, they will be initialized
 // at their first use with piInitialize.
-#define _PI_API(api) decltype(::api) * api = nullptr;
+#define _PI_API(api) decltype(::api) *api = nullptr;
 #include <CL/sycl/detail/pi.def>
 
-// TODO: implement real plugins (ICD-like?)
-// For now this has the effect of redirecting to built-in PI OpenCL plugin.
+// Find the plugin at the appropriate location and return the location in
+// PluginPath
+// TODO: Change the function appropriately when there are multiple plugins.
+bool findPlugin(std::string &PluginPath) {
```

Comment:
```suggestion
std::string findPlugin() {
```
and simply check if string is empty on calling point?

Reply:
The code will be more complex in this function.
It may not return a single plugin, but a set of plugins it finds. So we will be changing the API based on the functionality we add.  I think it is a little symmetrical to the bindPlugin API as well. Let me know if you feel it must be changed, I can do it.

Reply:
My first thought seeing the function in use (where the reference on the parameter is not visible) was that it searches for the plugin named in `PluginPath` and returns if it can find it.

Reply:
Makes sense. Ruyman's suggestion is more understandable in that case. 
I will change the function format.
done.